### PR TITLE
Api property endpoints

### PR DESCRIPTION
**Pull request by Jachin-Manda**
- add schema for Property db model and API URL endpoints for accessing the property data.

### CODE REVIEW

**Comment by Jachin-Manda** [comment]
I used the validate and ValidationError method that comes with Marshmallow in the Property schema to validate JSON data from a request. It provides a better way for validating data, so I will refactor code in the User schema as well and implement it. But I think it should be in another pull request as I don't want to mix the two.

File: api/endpoints/property.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        prop_data = property_schema.load(data)
+        prop_to_update.name = prop_data['name']
+        prop_to_update.desc = prop_data['desc']
+        prop_to_update.price = prop_data['price']
+        prop_to_update.location = prop_data['location']
+        prop_to_update.photos = prop_data['photos']
+        prop_to_update.date = datetime.utcnow()
+        # Checking the user_id foreign key if it points to the owner of the property.
+        if prop_to_update.user_id != int(prop_data['user_id']):
+            return jsonify({"error": f"The user with ID {prop_data['user_id']} is not the owner of the property"})
```
Comment (sonlinux) [inline api/endpoints/property.py:75]: what if I just mistyped a non-existing property ID?
maybe a more optimal approach would be to 

filter all properties based on the currently authenticated user (this calls for implementing authentication via API)
and then we return property from that filtered list or else we just say not found

File: api/endpoints/property.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        else:
+            prop_to_update.user_id = prop_data['user_id']
+        db.session.commit()
+        return jsonify({"message": "The details of the property has been updated."})
+    except ValidationError as err:
+        return jsonify(err.messages)
+
+
+@property_endpoint.route('/api/property/delete/<prop_id>', methods=['DELETE'])
+def delete_user(prop_id):
```
Comment (sonlinux) [inline api/endpoints/property.py:85]: is this `delete_user` or `delete_property`
A typo perhaps?

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: api/endpoints/property.py (hunk 2 above)
Comment (Jachin-Manda) [reply api/endpoints/property.py:85]: It should be `delete_property`. It's was just a typo

File: api/endpoints/property.py (hunk 1 above)
Comment (Jachin-Manda) [reply api/endpoints/property.py:75]: I will remove it then since this can be taken care of with API authentication.

**Pull request merged**